src/main/kotlin/no/nav/familie/ba/sak/kjerne/vilkårsvurdering/VilkårsvurderingForskyvningUtils.kt

```diff
@@ -113,15 +115,30 @@ object VilkårsvurderingForskyvningUtils {
         return this
             .filter { it.vilkårType == vilkår }
             .tilTidslinje()
-            .tilMånedFraMånedsskifteIkkeNull { innholdSisteDagForrigeMåned, innholdFørsteDagDenneMåned ->
-                when {
-                    vilkår == Vilkår.BOR_MED_SØKER && innholdFørsteDagDenneMåned.erDeltBosted() -> innholdSisteDagForrigeMåned
-                    !innholdSisteDagForrigeMåned.erOppfylt() -> innholdSisteDagForrigeMåned
-                    else -> innholdFørsteDagDenneMåned
+            .tilMånedFraMånedsskifte { innholdSisteDagForrigeMåned, innholdFørsteDagDenneMåned ->
+                if (innholdFørsteDagDenneMåned != null && innholdSisteDagForrigeMåned != null) {
+                    when {
+                        vilkår == Vilkår.BOR_MED_SØKER && innholdFørsteDagDenneMåned.erDeltBosted() -> innholdSisteDagForrigeMåned
+                        !innholdSisteDagForrigeMåned.erOppfylt() -> innholdSisteDagForrigeMåned
+                        else -> innholdFørsteDagDenneMåned
+                    }
+                } else {
+                    if (innholdFørsteDagDenneMåned == null && innholdSisteDagForrigeMåned.erEksplisittAvslagInnenforSammeMåned()
+                    ) {
+                        innholdSisteDagForrigeMåned
+                    } else {
+                        null
+                    }
```

Comment:
Blir dette det samme?
```suggestion
                } else if (innholdFørsteDagDenneMåned == null && innholdSisteDagForrigeMåned.erEksplisittAvslagInnenforSammeMåned()
                ) {
                    innholdSisteDagForrigeMåned
                } else {
                    null
                    
```

Reply:
ja det burde ha samme effekt :)